helm-chart/renku-gateway/templates/configmap.yaml

```
@@ -113,6 +121,12 @@ data:
         [http.middlewares.uiRedirect.redirectRegex]
           regex = "http://(.*)/entities/(.*)"
           replacement = "{{(include "gateway.protocol" .)}}://${1}/${2}"
+        
+        {{ if .Values.gitlabUrl }}
+        [http.middlewares.gitlabRedirect.redirectRegex]
+          regex = "(.*)/gitlab(.*)"
+          replacement = "{{ .Values.gitlabUrl }}${2}"
```

```suggestion
          replacement = "{{ trimSuffix "/" .Values.gitlabUrl }}${2}"
```

Not sure if a double slash would be a problem there or not. But I guess better to not take any chances. It could be that someone defines the `gitlabUrl` with a slash at the end in the values file.

And from the regex you used above $2 will be either blank or will start with a `/`.

---

Thanks! That is a great way of handling the possible double-slash scenario, which I also encountered when testing.